pyfr/backends/openmp/types.py

```diff
@@ -55,6 +56,10 @@ class OpenMPXchgMatrix(OpenMPMatrix, base.XchgMatrix): pass
 class OpenMPXchgView(base.XchgView): pass
 class OpenMPView(base.View): pass
 
+# Need a hashable wrapper for a list of kernels for the topological sort
+class OpenMPGroupNode:
```

Comment:
See if we can avoid this by using `id`.

Reply:
Is the latest commit more what you were thinking in terms of using `id`? If it is, I've only tested it quickly - I'll set off some more tests overnight and let you when they've finished before merging